perf: limit workspace project concurrency to available cores

sheremet-va opened the pull request:
### Description

<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->

<!-- You can also add additional context here -->

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

AriPerkkio reviewed (commented):
Hold on, what concurrency is this? Workspace projects are always run sequentially, right?

sheremet-va commented:
> Hold on, what concurrency is this? Workspace projects are always run sequentially, right?

No. It was running sequentially when we had to do the `process.chdir`. Before that, it was running in parallel so I reverted to the same behaviour when we removed the `process.chdir` hack. At the time I also added the concurrency limit of 5 just in case.
